src/reformat_libyuv.c

```diff
@@ -135,101 +135,68 @@ avifResult avifImageRGBToYUVLibYUV8bpc(avifImage * image, const avifRGBImage * r
-        }
-
-        if (!RGBtoY) {
-            return AVIF_RESULT_NOT_IMPLEMENTED;
-        }
-        if (RGBtoY(rgb->pixels, rgb->rowBytes, image->yuvPlanes[AVIF_CHAN_Y], image->yuvRowBytes[AVIF_CHAN_Y], image->width, image->height)) {
-            return AVIF_RESULT_REFORMAT_FAILED;
-        }
-        return AVIF_RESULT_OK;
-    }
-
-    // Generic mapping from any RGB layout (with or without alpha) to any YUV layout (subsampled or not).
-    int (*RGBtoYUV)(const uint8_t *, int, uint8_t *, int, uint8_t *, int, uint8_t *, int, int, int) = NULL;
-
     // libyuv only handles BT.601 for RGB to YUV, and not all range/order/subsampling combinations.
     // BT.470BG has the same coefficients as BT.601.
     if ((image->matrixCoefficients == AVIF_MATRIX_COEFFICIENTS_BT470BG) || (image->matrixCoefficients == AVIF_MATRIX_COEFFICIENTS_BT601)) {
-        if (rgb->format == AVIF_RGB_FORMAT_BGRA) {
-            if (image->yuvFormat == AVIF_PIXEL_FORMAT_YUV420) {
-                if (image->yuvRange == AVIF_RANGE_FULL) {
-                    RGBtoYUV = ARGBToJ420;
-                } else {
-                    RGBtoYUV = ARGBToI420;
-                }
-            } else if (image->yuvFormat == AVIF_PIXEL_FORMAT_YUV422) {
-                if (image->yuvRange == AVIF_RANGE_FULL) {
-                    RGBtoYUV = ARGBToJ422;
-                } else {
-                    RGBtoYUV = ARGBToI422;
-                }
-            } else if (image->yuvFormat == AVIF_PIXEL_FORMAT_YUV444) {
-                if (image->yuvRange == AVIF_RANGE_LIMITED) {
-                    RGBtoYUV = ARGBToI444;
-                }
-            }
-        } else if (rgb->format == AVIF_RGB_FORMAT_BGR) {
-            if (image->yuvFormat == AVIF_PIXEL_FORMAT_YUV420) {
-                if (image->yuvRange == AVIF_RANGE_FULL) {
-                    RGBtoYUV = RGB24ToJ420;
-                } else {
-                    RGBtoYUV = RGB24ToI420;
-                }
-            }
-        } else if (rgb->format == AVIF_RGB_FORMAT_RGBA) {
-            if (image->yuvFormat == AVIF_PIXEL_FORMAT_YUV420) {
-                if (image->yuvRange == AVIF_RANGE_FULL) {
-                    RGBtoYUV = avifABGRToJ420;
-                } else {
-                    RGBtoYUV = ABGRToI420;
-                }
-            }
-        } else if (rgb->format == AVIF_RGB_FORMAT_ARGB) {
-            if (image->yuvFormat == AVIF_PIXEL_FORMAT_YUV420) {
-                if (image->yuvRange == AVIF_RANGE_LIMITED) {
-                    RGBtoYUV = BGRAToI420;
+        if ((image->yuvFormat == AVIF_PIXEL_FORMAT_YUV444) || (image->yuvFormat == AVIF_PIXEL_FORMAT_YUV422) ||
+            (image->yuvFormat == AVIF_PIXEL_FORMAT_YUV420)) {
+            // Lookup table for RGB To YUV Matrix (average filter).
+            typedef int (*RGBtoYUV)(const uint8_t *, int, uint8_t *, int, uint8_t *, int, uint8_t *, int, int, int);
+            // Third dimension is for avifRange: 0 is AVIF_RANGE_LIMITED, 1 is AVIF_RANGE_FULL.
```

Comment:
Optional: The "0 is AVIF_RANGE_LIMITED, 1 is AVIF_RANGE_FULL" part could be omitted, because they are the numeric values of `AVIF_RANGE_LIMITED` and `AVIF_RANGE_FULL`.

Also at line 174.